tests/utils/test_ellipsoid_utils.py

```diff
@@ -2,17 +2,201 @@
+        points and 12 camera frustums from the door-12 dataset. Determines if the points are properly "locked" in
+        with one another before and after the alignment transformation is applied.
+        """
+        sample_data = GtsfmData(number_images=12)
+
+        # Instantiate OlssonLoader to read camera poses from door12 dataset.
+        wTi_list = self.loader._wTi_list
+
+        # Add 12 camera frustums to sample_data.
+        default_intrinsics = Cal3Bundler(fx=100, k1=0, k2=0, u0=0, v0=0)
+        for idx, pose in enumerate(wTi_list):
+            camera = PinholeCameraCal3Bundler(pose, default_intrinsics)
+            sample_data.add_camera(idx, camera)
+
+        # fmt: off
+        # These points are taken directly from the first 11 points generated by GTSFM on the door12 dataset (without
+        # any separate alignment transformation being applied)
+        points_3d = np.array(
+            [
+                [-1.4687794397729077, -1.4966178675020756, 14.583277665978546],
+                [-1.6172612359102505, -1.0951470733744013, 14.579095414379562],
+                [-3.2190882723771783, -4.463465966172758, 14.444076631000476],
+                [-0.6754206497590093, -1.1132530165104157, 14.916222213341355],
+                [-1.5514099044537981, -1.305810425894855, 14.584788688422206],
+                [-1.551319353347404, -1.304881682597853, 14.58246449772602],
+                [-1.9055918588057448, -1.192867982227922, 14.446379510423219],
+                [-1.5936792439193013, -1.4398818807488012, 14.587749795933021],
+                [-1.5937405395983737, -1.4401641027442411, 14.588167699143174],
+                [-1.6599318889904735, -1.2273604755959784, 14.57861988411431],
+                [2.1935589900444867, 1.6233406628428935, 12.610234497076608]
+            ]
+        )
+        # fmt: on
+
+        # Add all point cloud points to sample_data
+        for point_3d in points_3d:
+            sample_data.add_track(SfmTrack(point_3d))
+
+        camera_translations = np.array([pose.translation() for pose in sample_data.get_camera_poses()])
+        initial_relative_distances = self.compute_relative_distances(camera_translations, points_3d)
+
+        # Apply alignment transformation to sample_data
+        walignedTw = ellipsoid_utils.get_ortho_axis_alignment_transform(sample_data)
+        walignedTw = Similarity3(R=walignedTw.rotation(), t=walignedTw.translation(), s=1.0)
+        sample_data = sample_data.apply_Sim3(walignedTw)
+
+        # Aggregate the final, transformed points
+        num_tracks = sample_data.number_tracks()
+        transformed_points_3d = [np.array(sample_data.get_track(i).point3()) for i in range(num_tracks)]
+        transformed_points_3d = np.array(transformed_points_3d)
+        transformed_camera_translations = np.array([pose.translation() for pose in sample_data.get_camera_poses()])
+
+        final_relative_distances = self.compute_relative_distances(
```

Comment:
why only distances? do the points have a rigid transform relative to the cameras?